raspyrfm_client/device/base.py

```diff
@@ -35,7 +51,8 @@ def set_channel_config(self, **channel_arguments) -> None:
 
         :param channel_arguments:
         """
-        raise NotImplementedError
+        if self.check_channel_config(**channel_arguments):
+            self._channel = channel_arguments
```

Comment:
I wouldn't want to return true by default. A developer should always think about checking the configuration input when implementing a new device. This will also help Home Assistant users when debugging wrong configurations. If the developer does not want to check anything he can return true himself, but the implementation (and intent to check at least something) should be required.

The `get_channel_config_args` method is a good idea. It would be great if the configuration is specified in a way that allows for programmatic testing of all possible configurations (not for their result but at least successful execution). This would require the configuration keys, value types + value range to be programmatically available though and I don't know if this is possible in a simple, developer friendly way.

If I understand your first suggestion correctly it would replace the `_argchecks` field right? The automatic key + value range check is a great thing and I wouldn't want to drop this. Developers can override the `check_channel_config` method anyway if they need to.